File: PART2/ch01/ch-01 cookie/cookie/src/main/java/com/example/cookie/service/UserService.java
package com.example.cookie.service;

import com.example.cookie.db.UserRepository;
import com.example.cookie.model.LoginRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletResponse;

@RequiredArgsConstructor
@Service
public class UserService {

    private final UserRepository userRepository;

    // login logic
    public void login(
        LoginRequest loginRequest,
        HttpServletResponse httpServletResponse
    ){
        var id = loginRequest.getId();
        var pw = loginRequest.getPassword();

        var optionalUser = userRepository.findByName(id);

        if(optionalUser.isPresent()){
            var userDto = optionalUser.get();


            if(userDto.getPassword().equals(pw)){
                // cookie 해당 정보를 저장
                var cookie = new Cookie("authorization-cookie",userDto.getId());
                cookie.setDomain("localhost");  // naver.com , daum.net  dev.xxx.com, << production.xxx.com
                cookie.setPath("/");
                cookie.setHttpOnly(true);
                //cookie.setSecure(true); // << https 에서만 사용되도록 설정
                cookie.setMaxAge(-1);   // session

                httpServletResponse.addCookie(cookie);
            }
        }else{
            throw new RuntimeException("User Not Found");
        }

    }
}

File: PART3/ch02/ch02_01/java-kotlin/src/main/kotlin/org/example/user/UserService.kt
package org.example.user

import org.example.model.UserDto
import java.time.LocalDateTime

class UserService {

    fun logic(userDto: UserDto? = null){

        val userDto = UserDto(
            null,
            10,
            "gmail.com",
            "010-1111-2222",
            LocalDateTime.now()
        )

        userDto.name?.let { println(it.length) }?: println("null".length)
    }

}

fun main(){
    UserService().logic()
}